BootloaderCorePkg/Library/AcpiInitLib/AcpiInitLib.c

```diff
@@ -208,6 +292,22 @@ AcpiTableUpdate (
   while (Size < Length) {
     AcpiHdr = (EFI_ACPI_DESCRIPTION_HEADER *)(AcpiTable + Size);
 
+    // Update SBL Performance Table if signature matches
+    if (AcpiHdr->Signature == EFI_ACPI_5_0_FIRMWARE_PERFORMANCE_DATA_TABLE_SIGNATURE) {
+      // FPDT points to SBL Performance Table
+      Fpdt = (FIRMWARE_PERFORMANCE_TABLE *)AcpiHdr;
+      SblPerfTable = (SBL_PERFORMANCE_TABLE *)(UINTN)Fpdt->SblPerfPointerRecord.SblPerfTablePointer;
+      if (SblPerfTable == NULL) {
+        DEBUG((DEBUG_INFO, "SBL Performance table not found!\n"));
+        Status = EFI_NOT_FOUND;
+        break;
+      } else {
+        DEBUG ((DEBUG_INFO, "SBLT found @ 0x%X, updating..\n", SblPerfTable));
+        Status = UpdateAcpiSblt(SblPerfTable);
+        break;
+      }
+    }
+
```

Comment:
@gdong1 - Here I check the passed ACPI table signature and if it is a FPDT table, it is updated instead of appending a new table.

Reply:
Got it. you updated the AcpiTableUpdate () for FPDT.